src/networkwatcher.cpp

```diff
@@ -174,3 +175,31 @@ QString NetworkWatcher::getCurrentTransport() {
   return QString(metaEnum.valueToKey(type))
       .remove("TransportType_", Qt::CaseSensitive);
 }
+
+void NetworkWatcher::setImpl(NetworkWatcherImpl* networkWatcherImpl) {
+  logger.debug() << "setImpl";
+
+  m_impl = new InspectorNetworkWatcher(this);
+
+  connect(m_impl, &NetworkWatcherImpl::unsecuredNetwork, this,
+          &NetworkWatcher::unsecuredNetwork);
+  connect(m_impl, &NetworkWatcherImpl::networkChanged, this,
+          &NetworkWatcher::networkChange);
+
+  m_impl->initialize();
+
+  SettingsHolder* settingsHolder = SettingsHolder::instance();
+  Q_ASSERT(settingsHolder);
+
+  m_active = settingsHolder->unsecuredNetworkAlert() ||
+             settingsHolder->captivePortalAlert();
+  m_reportUnsecuredNetwork = settingsHolder->unsecuredNetworkAlert();
+  if (m_active) {
+    m_impl->start();
+  }
+
+  connect(settingsHolder, &SettingsHolder::unsecuredNetworkAlertChanged, this,
+          &NetworkWatcher::settingsChanged);
+  connect(settingsHolder, &SettingsHolder::captivePortalAlertChanged, this,
+          &NetworkWatcher::settingsChanged);
```

Comment:
It should be possible to remove this if Basti's suggestion of using setImpl for the other watchers is followed, because NetworkWatcher::initialize will provide this functionality.